### ISSUE TITLE
Sanity check on documents4j server

### ISSUE THREAD

**Issue by offwhitepanda** [documents4j/documents4j#144]
Good afternoon,
I am hosting my documents4j on a windows server and I am looking to see if there is a way to perform a sanity check on the doc4j server? Like a way I can tell the service is up and operational or if there is a log somewhere that will be created when there is an issue?
I apologize if this is not the correct place or way to ask this. Any assistance would be appreciated.

Background: we get an error when trying to send a conversion request from the client to the server. (Because something has gone awry). Restarting java and reloading the program fixes the issue but the only way i know is after a user has reported they couldnt convert.

**Comment by raphw** [comment]
There is a [liveliness check](https://github.com/documents4j/documents4j/blob/master/documents4j-api/src/main/java/com/documents4j/api/IConverter.java#L88), but it relies on Word's own reporting.

Alternatively, you can do a routine-check by converting a document that you know. If it does not work, Word has crashed.

**Comment by offwhitepanda** [comment]
Thank you for responding, I am a bit new to programming and usually use C# so please forgive me for my ignorance:
I see that you also have a mockconversion function. I am looking to essentially add something like:

import java.io.FileWriter;
import java.io.IOException;

public class LogFile {
    private static final String LOG_FILE_NAME = "\\document4j\\log.txt";
    
    public static void log(String message) {
        try (FileWriter writer = new FileWriter(System.getenv("SystemDrive") + LOG_FILE_NAME, true)) {
            writer.write(message + System.lineSeparator());
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

_______________________________________________________________________________________________________________________

public class MySanityCheck{

    public void SanityCheck(){
        IConverter myConverter = new IConverter(/*Insert Requirements for IConverter Here*/);
        bool status = myConverter.isOperational();
        if (!status){
            LogFile myLog = new LogFile("The isOperational check has failed. Word may not be responding.");
        }
        
        //Attempt mock conversion here
        //If the mock conversion fails -->
        LogFile log = new LogFile("The mock conversion check has failed.");
        
    }

}


_________________________________________________________________________________________________________________________

Essentially a very simple text log if the isOperational fails or a pretend conversion.
If there is any advice or assistance you can provide it would much be appreciated.

**Comment by raphw** [comment]
That's a rather general programming question that is not really related to documents4j.

Maybe you would rather want to use the command line tools?

**Comment by offwhitepanda** [comment]
That is correct i suppose i was just trying to implement what i thought you were indicating in your first response.
If there is a command line tool that will help i would love to know. I saw something about "/health" and "/running" to monitor the health. Is that a command line option or an http addition like http://0.0.0.0:9998/health ?

**Comment by raphw** [comment]
There's a standalone client and server jar that you can use with documentation on each.

**Comment by offwhitepanda** [comment]
I will reread the documentations. Thank you.
